### PR TITLE
Add document for @checkup/cli

### PR DESCRIPTION
**Pull request by zhanwang626**
#1 

- Add CLI in API sidebar
- Add `generate-docs.js`
- Generate documentations for `@checkup/cli`

## Preview
image

image

### CODE REVIEW

File: scripts/generate-docs.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
+const path = require('path');
```
Comment (scalvert) [inline scripts/generate-docs.js:1]: I think we can probably use some stuff in `readme-api-generator`.

File: scripts/generate-docs.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      filePathList.push(path.resolve(dirName, relativePath[1]));
+    }
+  });
+
+  return filePathList;
+}
+
+(async function () {
+  const rootDir = path.dirname(__dirname);
+  const filePathList = getFilePathList(path.resolve(rootDir, 'node_modules/@checkup/cli/lib/index.d.ts'));
```
Comment (scalvert) [inline scripts/generate-docs.js]: We should definitely not hardcode this. Let's try to use some dynamic stuff from `readme-api-generator`.

**Review by scalvert — CHANGES_REQUESTED** [review]

File: scripts/generate-docs.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  const docsPlaceholder = /<!--DOCS_START-->[\S\s]*<!--DOCS_END-->/;
+
+  for (let filePath of filePathList) {
+    const fileName = filePath.replace(/^.*[\\\/]/, '');
+    const mdFile = path.resolve(rootDir, 'docs/api/cli', `${fileName}.md`);
+    const mdContent = fs.readFileSync(mdFile, 'utf8');
```
Comment (scalvert) [inline scripts/generate-docs.js]: I think this assumes that the file exists already? We should probably auto-generate the files, and overwrite if necessary.

File: scripts/generate-docs.js (hunk 1 above)
Comment (zhanwang626) [reply scripts/generate-docs.js:1]: While I didn't find any stuff is useful for our use case...

Comment (scalvert) [reply scripts/generate-docs.js:1]: OK. We should discuss, since part of the reason I wrote it was to make it easier to generate docs.

**Review by scalvert — APPROVED** [review]

**Pull request merged**